File: Cplusplus/최소직사각형.cpp
//22.01.13 최소직사각형 by CJHofficial

#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int maxWidth = 0;
    int maxHeight = 0;
    for (auto v : sizes) {
        sort(v.begin(), v.end());
        if (v[0] > maxWidth) maxWidth = v[0];
        if (v[1] > maxHeight) maxHeight = v[1];
    }
    answer = maxWidth * maxHeight;
    return answer;
}

File: 최소직사각형.py
def solution(sizes):
    answer = [0,0]
    for array in sizes:
        array.sort()
        answer[0] = answer[0] if answer[0] > array[0] else array[0]
        answer[1] = answer[1] if answer[1] > array[1] else array[1]
    return answer[0] * answer[1]
